lib/polymorphic_integer_type/belongs_to_polymorphic_association_extension.rb

```diff
@@ -3,7 +3,9 @@ module Associations
     class BelongsToPolymorphicAssociation < BelongsToAssociation
       private def replace_keys(record)
         super
-        owner[reflection.foreign_type] = record.class
+        unless record.nil?
+          owner[reflection.foreign_type] = record.class.base_class
```

Comment:
@fimmtiu I think it makes sense to do nothing if record is nil to avoid potential side effects.